ceilometer_publisher_zeromq/queue.py

```diff
@@ -0,0 +1,55 @@
+from ceilometer import publisher
+from ceilometer.openstack.common.gettextutils import _
+from ceilometer.openstack.common import log
+from oslo.config import cfg
+import json
+import zmq
+
+LOG = log.getLogger(__name__)
+
+OPTS = [
+    cfg.StrOpt('publisher_zeromq_host',
+               default='127.0.0.1'),
+    cfg.IntOpt('publisher_zeromq_port',
+               default=8282),
+]
+
+cfg.CONF.register_opts(OPTS)
+
+class QueuePublisher(publisher.PublisherBase):
+    """Republishes all received samples to a collector via ZeroMQ"""
+
+    def __init__(self, parsed_url):
+        LOG.info("Queue Publisher starting up")
+        super(QueuePublisher, self).__init__(parsed_url)
+        self.context = None
+        self.socket  = None
+        self.reconnect()
+
+    def reconnect(self):
+        """(re)connects to the collector"""
+        self.context = zmq.Context()
```

Comment:
I wasn't sure if zmq_disconnect would reset the state of a REQ/REP socket pair. It probably does though.